tools/riemann-chronos/bin/riemann-chronos

```diff
@@ -0,0 +1,145 @@
+#!/usr/bin/env ruby
+
+require 'riemann/tools'
+
+class Riemann::Tools::Chronos
+  include Riemann::Tools
+
+  require 'faraday'
+  require 'json'
+  require 'uri'
+
+  opt :read_timeout, 'Faraday read timeout', type: :int, default: 2
+  opt :open_timeout, 'Faraday open timeout', type: :int, default: 1
+  opt :path_prefix, 'Chronos path prefix for proxied installations e.g. "chronos" for target http://localhost/chronos/metrics', default: "/"
+  opt :chronos_host, 'Chronos host', default: "localhost"
+  opt :chronos_port, 'Chronos port', type: :int, default: 4400
+
+  def initialize
+    options[:interval] = 60
+    options[:ttl] = 120
+  end
+
+  # Handles HTTP connections and GET requests safely
+  def safe_get(uri)
+      # Handle connection timeouts
+      response = nil
+      begin
+        connection = Faraday.new(uri)
+        response = connection.get do |req|
+          req.options[:timeout] = options[:read_timeout]
+          req.options[:open_timeout] = options[:open_timeout]
+        end
+      rescue => e
+        report(:host => uri.host,
+          :service => "chronos health",
+          :state => "critical",
+          :description => "HTTP connection error: #{e.class} - #{e.message}"
+        )
+      end
+      response
+  end
+
+  def health_url
+    path_prefix = options[:path_prefix]
+    path_prefix[0] = '' if path_prefix[0]=='/'
+    path_prefix[path_prefix.length-1] = '' if path_prefix[path_prefix.length-1]=='/'
+    "http://#{options[:chronos_host]}:#{options[:chronos_port]}#{path_prefix.length>0?'/':''}#{path_prefix}/metrics"
+  end
+
+  def jobs_url
+    path_prefix = options[:path_prefix]
+    path_prefix[0] = '' if path_prefix[0]=='/'
+    path_prefix[path_prefix.length-1] = '' if path_prefix[path_prefix.length-1]=='/'
+    "http://#{options[:chronos_host]}:#{options[:chronos_port]}#{path_prefix.length>0?'/':''}#{path_prefix}/scheduler/jobs"
+  end
+
+  def tick
+    tick_health
+    tick_jobs
+  end
+
+  def tick_health
+    uri = URI(health_url)
+    response = safe_get(uri)
+
+    return if response.nil?
+
+    if response.status != 200
+        report(:host => uri.host,
+          :service => "chronos health",
+          :state => "critical",
+          :description => "HTTP connection error: #{response.status} - #{response.body}"
+        )
+    else
+      # Assuming that a 200 will give json
+      json = JSON.parse(response.body)
+      state = "ok"
```

Comment:
Yes, you're right.  I was trying to keep the initial delta between riemann-marathon and riemann-chronos as small as possible.

```text
diff -uw tools/riemann-marathon/bin/riemann-marathon tools/riemann-chronos/bin/riemann-chronos
```

```diff
--- tools/riemann-marathon/bin/riemann-marathon	2016-03-15 10:53:16.000000000 +1100
+++ tools/riemann-chronos/bin/riemann-chronos	2016-03-15 13:02:28.000000000 +1100
@@ -2,7 +2,7 @@

 require 'riemann/tools'

-class Riemann::Tools::Marathon
+class Riemann::Tools::Chronos
   include Riemann::Tools

   require 'faraday'
@@ -11,9 +11,9 @@

   opt :read_timeout, 'Faraday read timeout', type: :int, default: 2
   opt :open_timeout, 'Faraday open timeout', type: :int, default: 1
-  opt :path_prefix, 'Marathon path prefix for proxied installations e.g. "marathon" for target http://localhost/marathon/metrics', default: "/"
-  opt :marathon_host, 'Marathon host', default: "localhost"
-  opt :marathon_port, 'Marathon port', type: :int, default: 8080
+  opt :path_prefix, 'Chronos path prefix for proxied installations e.g. "chronos" for target http://localhost/chronos/metrics', default: "/"
+  opt :chronos_host, 'Chronos host', default: "localhost"
+  opt :chronos_port, 'Chronos port', type: :int, default: 4400

   def initialize
     options[:interval] = 60
@@ -32,7 +32,7 @@
         end
       rescue => e
         report(:host => uri.host,
-          :service => "marathon health",
+          :service => "chronos health",
           :state => "critical",
           :description => "HTTP connection error: #{e.class} - #{e.message}"
         )
@@ -44,19 +44,19 @@
     path_prefix = options[:path_prefix]
     path_prefix[0] = '' if path_prefix[0]=='/'
     path_prefix[path_prefix.length-1] = '' if path_prefix[path_prefix.length-1]=='/'
-    "http://#{options[:marathon_host]}:#{options[:marathon_port]}#{path_prefix.length>0?'/':''}#{path_prefix}/metrics"
+    "http://#{options[:chronos_host]}:#{options[:chronos_port]}#{path_prefix.length>0?'/':''}#{path_prefix}/metrics"
   end

-  def apps_url
+  def jobs_url
     path_prefix = options[:path_prefix]
     path_prefix[0] = '' if path_prefix[0]=='/'
     path_prefix[path_prefix.length-1] = '' if path_prefix[path_prefix.length-1]=='/'
-    "http://#{options[:marathon_host]}:#{options[:marathon_port]}#{path_prefix.length>0?'/':''}#{path_prefix}/v2/apps"
+    "http://#{options[:chronos_host]}:#{options[:chronos_port]}#{path_prefix.length>0?'/':''}#{path_prefix}/scheduler/jobs"
   end

   def tick
     tick_health
-    tick_apps
+    tick_jobs
   end

   def tick_health
@@ -67,7 +67,7 @@

     if response.status != 200
         report(:host => uri.host,
-          :service => "marathon health",
+          :service => "chronos health",
           :state => "critical",
           :description => "HTTP connection error: #{response.status} - #{response.body}"
         )
@@ -77,14 +77,14 @@
       state = "ok"

       report(:host => uri.host,
-             :service => "marathon health",
+             :service => "chronos health",
              :state => state)

       json.each_pair do |t, d|
         if d.respond_to? :each_pair
           d.each_pair do |service, counters|
             report(:host => uri.host,
-                   :service => "marathon_metric #{t} #{service}",
+                   :service => "chronos_metric #{t} #{service}",
                    :metric => 1,
                    :tags => ["metric_name"],
                    :ttl => 600
@@ -93,7 +93,7 @@
               counters.each_pair do |k, v|
                 if v.is_a? Numeric
                   report(:host => uri.host,
-                         :service => "marathon #{service} #{k}",
+                         :service => "chronos #{service} #{k}",
                          :metric => v,
                          :tags => ["metric", "#{t}"],
                          :ttl => 600
@@ -107,15 +107,15 @@
     end
   end

-  def tick_apps
-    uri = URI(apps_url)
+  def tick_jobs
+    uri = URI(jobs_url)
     response = safe_get(uri)

     return if response.nil?

     if response.status != 200
         report(:host => uri.host,
-          :service => "marathon health",
+          :service => "chronos health",
           :state => "critical",
           :description => "HTTP connection error: #{response.status} - #{response.body}"
         )
@@ -125,14 +125,14 @@
       state = "ok"

       report(:host => uri.host,
-             :service => "marathon health",
+             :service => "chronos health",
              :state => state)

-      json["apps"].each do |app|
-        app.each_pair do |k, v|
+      json.each do |job|
+        job.each_pair do |k, v|
           if v.is_a? Numeric
             report(:host => uri.host,
-                   :service => "marathon apps#{app["id"]}/#{k}",
+                   :service => "chronos job #{job["name"]} #{k}",
                    :metric => v,
                    :ttl => 120
             )
@@ -142,5 +142,4 @@
     end
   end
 end
-Riemann::Tools::Marathon.run
-
+Riemann::Tools::Chronos.run
```